tests/integration/pilot/common/routing.go

```diff
@@ -155,13 +157,11 @@ spec:
 			opts: echo.CallOptions{
 				PortName: "http",
 				Count:    1,
-				Validator: echo.And(
-					echo.ExpectOK(),
-					echo.ValidatorFunc(
-						func(response echoclient.ParsedResponses, _ error) error {
-							return response.Check(func(_ int, response *echoclient.ParsedResponse) error {
-								return ExpectString(response.RawResponse["Istio-Custom-Header"], "user-defined-value", "request header")
-							})
+				Checker: check.And(
+					check.OK(),
+					check.Func(
+						func(r *echoclient.Instance) error {
+							return ExpectString(r.RawResponse["Istio-Custom-Header"], "user-defined-value", "request header")
```

Comment:
nit: could use check.Key

Reply:
that does lose the "help" bit.

Options:
* Just lose the help bit
* Add new check.WithHelp(check.Key(foo, bar), "some help message")
* Make ExpectString the same but return a check.Func so we don't always need to wrap this two levels deep.

With some of the other comments on "check.OK" wthis would look like `Checker: check.Key("Istio-Custom-Header", "user-defined-value")` or similar which is super clean